strapi/strapi
# Warn about incorrect node version on Strapi load

### What does it do?

- move the check-requirements from create-strapi-app to @strapi/utils
- use check-requirements in Strapi load to emit an error to users if they're loading Strapi with the wrong version of node

### Why is it needed?

We currently only check node version on new project creation. The "engines" field in package.json is not enforced and can't be relied on by itself.

### How to test it?

Start strapi
- if you have an unsupported node version, you should see an error or warning logged but it should attempt to proceed with server startup
- if you have a supported node version, Strapi should start up as normal without warning

Use create-strapi-app to create a new project
- if you have an unsupported node version, you should see an error and it should not proceed.
- if you have a supported node version, it should still work

### TODO

- [ ] we should return error messages or codes instead of logging directly
- [ ] decide if @strapi/utils is an appropriate place for this, or if it should remain duplicated in create-strapi-app to reduce size and dependencies

### Related issue(s)/PR(s)

Let us know if this is related to any issue/pull request
